File: frontend/src/types/jobRequirement.ts
export type JobRequirementStatus = 'open' | 'on_hold' | 'closed' | 'filled'
export type OwnerStatusType = 'pending_review' | 'approved' | 'rejected'

export interface SubmissionField {
  field: string
  type: string
  required: boolean
}

export interface CandidateSubmission {
  id: string
  candidate_id: string
  candidate_name: string
  candidate_email: string
  job_requirement_id: string
  job_title: string
  client_id: string
  client_name: string
  submitted_by_id: string | null
  submitted_by_name: string | null
  submitted_at: string | null
  current_stage: string | null
  status: string
  owner_status: OwnerStatusType
  in_pipeline?: boolean
  submission_data: Record<string, string> | null
}

export interface OwnerDashboardJob {
  job_requirement_id: string
  job_title: string
  status: string
  open_positions: number | null
  submissions: CandidateSubmission[]
}

export interface OwnerDashboardClient {
  client_id: string
  company_name: string
  jobs: OwnerDashboardJob[]
}

export interface JobRequirementListItem {
  id: string
  client_id: string
  client_name: string
  assigned_to: string | null
  assigned_recruiter_name: string | null
  job_title: string
  department?: string | null
  employment_type?: string | null
  work_mode?: string | null
  experience_min?: number | null
  experience_max?: number | null
  salary_min?: number | null
  salary_max?: number | null
  open_positions?: number | null
  job_description?: string | null
  location?: string | null
  priority?: string | null
  status: JobRequirementStatus | string
  requirement_type?: string | null
  created_by: string
  created_at?: string | null
  client_submission_format?: SubmissionField[] | null
}

export interface JobRequirementCreatePayload {
  client_id: string
  assigned_to: string
  job_title: string
  department?: string | null
  employment_type?: string | null
  work_mode?: string | null
  experience_min?: number | null
  experience_max?: number | null
  salary_min?: number | null
  salary_max?: number | null
  open_positions?: number | null
  job_description?: string | null
  location?: string | null
  priority?: string | null
  requirement_type?: string | null
  status?: JobRequirementStatus
}

export type JobRequirementUpdatePayload = Partial<JobRequirementCreatePayload>

File: backend/app/api/schemas/job_requirement.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import Any, Literal

from pydantic import BaseModel, Field


class JobRequirementCreateRequest(BaseModel):
    client_id: uuid.UUID
    assigned_to: uuid.UUID
    job_title: str = Field(min_length=1, max_length=255)
    department: str | None = None
    employment_type: str | None = None
    work_mode: str | None = None
    experience_min: Decimal | None = None
    experience_max: Decimal | None = None
    salary_min: Decimal | None = None
    salary_max: Decimal | None = None
    open_positions: int | None = Field(default=None, ge=1)
    job_description: str | None = None
    location: str | None = None
    priority: str | None = None
    requirement_type: str | None = None
    status: Literal["open", "on_hold", "closed", "filled"] = "open"


class JobRequirementUpdateRequest(BaseModel):
    client_id: uuid.UUID | None = None
    assigned_to: uuid.UUID | None = None
    job_title: str | None = Field(default=None, min_length=1, max_length=255)
    department: str | None = None
    employment_type: str | None = None
    work_mode: str | None = None
    experience_min: Decimal | None = None
    experience_max: Decimal | None = None
    salary_min: Decimal | None = None
    salary_max: Decimal | None = None
    open_positions: int | None = Field(default=None, ge=1)
    job_description: str | None = None
    location: str | None = None
    priority: str | None = None
    requirement_type: str | None = None
    status: Literal["open", "on_hold", "closed", "filled"] | None = None


class JobRequirementStatusUpdate(BaseModel):
    status: Literal["open", "on_hold", "closed", "filled"]


class JobRequirementListItem(BaseModel):
    id: uuid.UUID
    client_id: uuid.UUID
    client_name: str
    assigned_to: uuid.UUID | None = None
    assigned_recruiter_name: str | None = None
    job_title: str
    department: str | None = None
    employment_type: str | None = None
    work_mode: str | None = None
    experience_min: Decimal | None = None
    experience_max: Decimal | None = None
    salary_min: Decimal | None = None
    salary_max: Decimal | None = None
    open_positions: int | None = None
    job_description: str | None = None
    location: str | None = None
    priority: str | None = None
    status: str
    requirement_type: str | None = None
    created_by: uuid.UUID
    created_at: datetime | None = None
    client_submission_format: list[dict[str, Any]] | None = None
    submitted_candidates: int = 0
    pipeline_candidates: int = 0
    joined_candidates: int = 0
    remaining_positions: int | None = None
    submissions_enabled: bool = True
